include/drivers/irq/hifive.h

```diff
@@ -36,12 +36,19 @@
 #define PLIC_THRES_PER_CONTEXT  0x1000
 #define PLIC_THRES_CLAIM        0x4
 
-#ifdef CONFIG_PLAT_HIFIVE
+#if defined(CONFIG_PLAT_HIFIVE) || defined(CONFIG_PLAT_POLARFIRE)
```

Comment:
@Jesse-Millwood, I'm just wondering why does the PolarFire SoC have the same PLIC hardware register interface as the HiFive? 

The seL4 PLIC driver is currently structured with the generic interrupt handling functionality in src/arch/riscv/machine/hardware.c and then each platform providing its own header file for implementing the register accesses.  Does the HiFive Unleashed and PolarFire SoC share the same PLIC implementation or are the PLIC hardware register interfaces becoming standardized? I'm thinking that this file should be given a better name if it is no longer going to just be for the HiFive.

Reply:
I believe they do share the same PLIC implemntation. I'm pretty sure that Micro{chip,semi} used SiFive's u540 cores and PLIC. I can break this out into a separate mpfs.h if you'd like, it just seemed like there would be a lot of duplication.

Reply:
The PolarFire SoC is a continuation of the expansion board that they had made for the HiFive Unleashed [1]. UG0880 talks about using a CPU Core Complex [2], which I believe is the same as the SiFive U54-MC Core Complex [3]. Where the PLIC looks about the same in both. Specifically Table 57 in [2] and Table 48 in [3]


1: https://www.crowdsupply.com/microchip/polarfire-soc-icicle-kit
2: https://www.microsemi.com/document-portal/doc_download/1244570-ug0880-polarfire-soc-fpga-microprocessor-subsystem-mss-user-guide
3: https://sifive.cdn.prismic.io/sifive/6125646b-2536-463d-ba76-b77d8f344c53_sifive_U54MC_rtl_full_20G1.03.00_manual.pdf

Reply:
There's an [open source Chisel implementation of a PLIC](https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/devices/tilelink/Plic.scala) which is AFAIK used by all SiFive devices, as well as non-SiFive users of the rocket-chip platform.  Dunno how you would best express that.

Reply:
Maybe for now we could just rename it to riscv_plic0.h which maps the device tree compatibility string. I think it is fair to assume that devices declaring the same compatibility string can be supported by the same driver until we see otherwise.

Reply:
I made this change and pushed it.